### ISSUE TITLE
mock-task can't load task inputs and variables when AGENT_TEMPDIRECTORY is set

### ISSUE THREAD

**Issue by brcrista** [microsoft/azure-pipelines-task-lib#372]
Apparently from this change: https://github.com/Microsoft/vsts-task-lib/blame/master/node/internal.ts#L692

When tasks have tests that set a value for AGENT_TEMPDIRECTORY, the `getInput` function on mock-task (which is the same as the production version and calls into that _loadData function) is unable to load any of the task inputs.

**Repro**
install version >= 2.1 of task-lib to one of the following tasks and run the tests:
* DotNetCoreCLIV2
* InstallAppleCertificateV2
* InstallSSHKeyV0

**Comment by damccorm** [comment]
From #285 this also is an issue with `getVariable`

**Comment by tillig** [comment]
Ack! I think I just hit this while trying to write tests for a task that uses the `azure-pipelines-tool-lib/tool` library. It needs to download things, so temp directory needs to be set, but if I set that, then the input variable I passed in isn't found. I noticed the [standard Azure Pipelines tasks that download stuff](https://github.com/microsoft/azure-pipelines-tasks) either don't have tests or have _exceptionally convoluted_ mocking setups; I'm guessing this is why.
